backoffice/tests/UnitTests/Helpers/SearchHelpers/DroitQueryBuilderUnitTests.cs

```diff
@@ -0,0 +1,673 @@
+                ServicesEstimatedCost = 10, SalvageClaimAwarded = 6,
+                Salvor = salvor
+            },
+        }.AsQueryable();
+    
+        // Act
+        var result = DroitQueryBuilder.BuildQuery(form, droits, false);
+    
+        // Assert
+        Assert.Equal(1, result.Count()); 
+        Assert.True(result.Any(d => d.Reference == "MatchingDroit"));
+        Assert.False(result.Any(d => d.Reference == "NotMatchingDroit"));
+        Assert.False(result.Any(d => d.Reference == "OnlyPartialMatchingDroit"));
+    }
+
+    [Fact]
+    public void BuildQuery_WithSalvageSearchBooleanFields_ReturnsFilteredQuery()
+    {
+        // Arrange
+        var form = new DroitSearchForm
+        {
+            SalvageAwardClaimed = true,MMOLicenceRequired = true, MMOLicenceProvided = false
+        };
+        var droits = new List<Droit>
+        {
+            new()
+            {
+                Id = Guid.NewGuid(), Reference = "MatchingDroit",
+                SalvageAwardClaimed = true, MMOLicenceRequired = true, MMOLicenceProvided = false
+            },
+            new()
+            {
+                Id = Guid.NewGuid(), Reference = "NotMatchingDroit",
+                SalvageAwardClaimed = true, MMOLicenceRequired = true, MMOLicenceProvided = true
+            },
+        }.AsQueryable();
+    
+        // Act
+        var result = DroitQueryBuilder.BuildQuery(form, droits, false);
+    
+        // Assert
+        Assert.Equal(1, result.Count()); 
+        Assert.True(result.Any(d => d.Reference == "MatchingDroit"));
+        Assert.False(result.Any(d => d.Reference == "NotMatchingDroit"));
+    }
+    
+    [Fact]
+    public void BuildQuery_WithLegacySearchFields_ReturnsFilteredQuery()
+    {
+        // Arrange
+        var form = new DroitSearchForm
+        {
+            District = "Test" , LegacyFileReference = "Test", GoodsDischargedBy = "Test",
+            DateDelivered = "Test", Agent = "Test", RecoveredFromLegacy = "Test", ImportedFromLegacy = true
+        };
+        var salvor = new Salvor();
+        var droits = new List<Droit>
+        {
+            new()
+            {
+                Id = Guid.NewGuid(), Reference = "MatchingDroit",
+                District = "Test" , LegacyFileReference = "Test", GoodsDischargedBy = "Test",
+                DateDelivered = "Test", Agent = "Test", RecoveredFromLegacy = "Test", ImportedFromLegacy = true,
+                Salvor = salvor
+            },
+            new()
+            {
+                Id = Guid.NewGuid(), Reference = "PartiallyMatchingDroit",
+                District = "TestAndStuff" , LegacyFileReference = "Test", GoodsDischargedBy = "Test",
+                DateDelivered = "Test", Agent = "TestAndStuff", RecoveredFromLegacy = "Test", ImportedFromLegacy = true,
+                Salvor = salvor
+            },
+            new()
+            {
+                // if District is removed from droit it returns null or raises an exception?
```

Comment:
Probably worth making it a nullable string, it's a legacy field so no need for it to be required